### PR TITLE
WIP: sanity for request/show page

### PR DESCRIPTION
**Pull request by coolo**
I'll experiment with this page whenever I have time - hope noone else does :)

- [x] Merge comments and reviews
- [ ] Create diff page
- [ ] Add build status to diff page
- [ ] Display auto accept infos
- [ ] Readd forwarding
- [ ] Show package maintainer infos
- [ ] Add review button
- [ ] Mentioned issues (in action)

### CODE REVIEW

**Comment by bgeuken** [comment]
Not for the next two weeks. Happy hacking

**Comment by coolo** [comment]
the main idea is moving the actions diff in a tab and move the conversation/review into the middle of the focus - let's see how far I come with this approach (and the webui/webui2 split)

**Comment by mdeniz** [comment]
I added some tags...
Is this a work in progress then? If so, edit the title and prepend it with WIP to avoid the merge :+1:
If you feel is in a "good" shape then add a screenshot too.
And many thanks BTW!

**Comment by bgeuken** [comment]
> @coolo commented an hour ago
> the main idea is moving the actions diff in a tab and move the conversation/review into the middle of the focus - let's see how far I come with this approach (and the webui/webui2 split)

You probably know, but just in case... be aware that it's possible to have requests for multiple packages (with one tab for each package).

**Comment by coolo** [comment]
Yes, but I won't create tabs for actions. Just put them below each other. These tabs for actions do not scale. Just look at a random kernel update for sle

**Comment by hennevogel** [comment]
@coolo sorry, but re-designing workflows/pages doesn't make too much sense right now. We are going to come out with a gazillion small changes to the UI that people have to adopt to first. After that we can dive into workflow changes.

File: src/api/.haml-lint.yml [hunk 1]
```diff
@@ -28,6 +28,7 @@ linters:
       - Metrics/BlockNesting
       - Metrics/LineLength
       - Naming/FileName
+      - Rails/OutputSafety
```
Comment (coolo) [inline src/api/.haml-lint.yml]: @Ana06 if you know another way to shutup that cop, let me know

Comment (dmarcoux) [reply src/api/.haml-lint.yml]: I also had to "fight" with this lately, [this page](https://www.rubydoc.info/gems/rubocop/RuboCop/Cop/Rails/OutputSafety) helped me win the fight!

Comment (coolo) [reply src/api/.haml-lint.yml]: that worked

**Comment by coolo** [comment]
Reimplementing that crap view in bootstrap doesn't make sense either

**Comment by hennevogel** [comment]
> Reimplementing that crap view in bootstrap doesn't make sense either

I just told you why it is, sorry that you don't like this reasoning ¯\_(ツ)_/¯

**Comment by coolo** [comment]
First level unlocked! rendering reviews in a human readable way:

screencapture-e120-suse-de-3000-request-show-645670-2018-11-16-15_51_31

**Comment by vpereira** [comment]
@hennevogel @coolo as we are going to do bootstrap next sprint, does it make sense to refresh this PR and check if it still relevant?

**Comment by coolo** [comment]
what do you mean with refresh? It was updated 4 days ago

**Comment by vpereira** [comment]
> what do you mean with refresh? It was updated 4 days ago

well its updated but broken, not sure if the stuff you are doing are inline with what we are planing.. refresh: code and ideas, that's what I mean :)

**Comment by coolo** [comment]
It's WIP - and I only work on this when I'm blocked on my regular tasks.

**Comment by hennevogel** [comment]
> does it make sense to refresh this PR and check if it still relevant?

As explained previously, we try to keep workflow changes out of the migration. This is the :heart:  _THE_ OBS collaboration workflow, so no it does not  make sense :cowboy_hat_face:

File: src/api/app/controllers/webui/request_controller.rb [hunk 2]
```diff
@@ -79,7 +115,46 @@ def modify_review
...
+        reviews_seen << event.id
+      end
+      if event.is_a?(HistoryElement::RequestReviewAdded)
+        next if reviews_seen.include?(event.description_extension.to_i)
+        reviews_seen << event.description_extension.to_i
+      end
+      @events << event
+    end
+  end
+
```
Comment (vpereira) [inline src/api/app/controllers/webui/request_controller.rb]: why a method in the controller? Could you put in a service object  (i.e EventSorterService) and write tests for it?

File: src/api/app/views/webui2/webui/request/show.html.haml [hunk 3]
```diff
@@ -0,0 +1,68 @@
+:ruby
+  @pagetitle = "Request #{@bs_request.number}"
+  @layouttype = 'custom'
+
+.card.mb-3
+  = render partial: 'tabs', locals: { request: @bs_request }
+  .card-body
+    %h3
+      Request #{@bs_request.number} (#{@bs_request.state})
```
Comment (vpereira) [inline src/api/app/views/webui2/webui/request/show.html.haml]: you will probably have to adapt this code after the current deliverable

Comment (coolo) [reply src/api/app/views/webui2/webui/request/show.html.haml]: Sure

**Comment by vpereira** [comment]
Hi @coolo do you want to rebase it with master or close it?

**Comment by coolo** [comment]
I don't want to rebase it - it's hopeless with all the changes that happened in both sides. I want to cherry pick the ideas, so please leave it open

**Comment by vpereira** [comment]
> I don't want to rebase it - it's hopeless with all the changes that happened in both sides. I want to cherry pick the ideas, so please leave it open

The PRs aren't here for stuff that we won't merge. Could we close it and you keep your local branch for cherry pick the ideas when it is necessary?

**Comment by coolo** [comment]
ok, let me rephrase: I will update the PR, but not with a rebase but with a fresh approach

**Comment by coolo** [comment]
saved most but not all yet in https://github.com/openSUSE/open-build-service/pull/7591 - but I can close this one now
